src/osd/osd_types.cc

```diff
@@ -5813,10 +5813,9 @@ void OSDSuperblock::generate_test_instances(list<OSDSuperblock*>& o)
 
 void SnapSet::encode(ceph::buffer::list& bl) const
 {
-  ENCODE_START(3, 2, bl);
+  ENCODE_START(4, 2, bl);
   encode(seq, bl);
   encode(true, bl);  // head_exists
-  encode(snaps, bl);
```

Comment:
Oops, this would blow up, I'm afraid. Try to imagine a v3 decoder trying to unconditionally `decode()` the part of bytestream that is no there anymore.

I think an empty instance would need to be encoded just for the sake of compatibility.

Reply:
If we encode unconditionally, decode should also support the empty instance encoded.
Does this require a version bump up?

Reply:
@rzarzynski, I updated the commit. Please take a look.

Reply:
After bumping the version we don't need the snaps encode anymore, the decode function will handle it

Reply:
I don't understand what's happening here.  If we're going to encode snaps anyway, I don't think we need the version bump.  If we're going to remove the field, we'd need to bump the version *and* make it a feature-full encoder called with the minimum features any future osd is allowed to have.  Until the minimum osd version supports the new snaps-less encoding, we'll need to encode an empty one as here.